### PR TITLE
Node fun test improvements

### PR DESCRIPTION
**Pull request by fviernau**
Re-arrange the test assets for `node` and `yarn`. 

Note: Another PR will follow which makes analog changes for `pnpm` and `yarn2`.

### CODE REVIEW

File: plugins/package-managers/node/src/funTest/kotlin/NpmFunTest.kt [hunk 1]
```diff
@@ -40,7 +40,8 @@ import org.ossreviewtoolkit.utils.test.patchExpectedResult
 
 class NpmFunTest : WordSpec({
     "NPM" should {
-        "resolve shrinkwrap dependencies correctly" {
+        "resolve dependencies for a project with a 'shrinkwrap.json' correctly" {
+            // See also: https://docs.npmjs.com/cli/v6/configuring-npm/shrinkwrap-json.
```
Comment (sschuberth) [inline plugins/package-managers/node/src/funTest/kotlin/NpmFunTest.kt:44]: Actually, I'm not sure that the comment is useful here. In what regard should I look at the description of a `npm-shrinkwrap.json` in the context of this test?

IMO such a link would be more useful in the code under test that deals with the `npm-shrinkwrap.json` file, like the `NodePackageManager` enum.

TL;DR I'd just remove the `README.md`, and optionally document the enums in `NodePackageManager` in another commit with references to the mentioned files.

And while at it a commit title nit: `README.md` could be put in backticks.

**Review by sschuberth — COMMENTED** [review]
Are you also planning to move Yarn2 and PNPM test assets below a common base directory?

**Comment by fviernau** [comment]
> Are you also planning to move Yarn2 and PNPM test assets below a common base directory?

Yes, I've explained this in the PR description.

**Comment by sschuberth** [comment]
> Yes, I've explained this in the PR description.

Sorry, I need to get used to reading these in addition to the commit messages.

File: plugins/package-managers/node/src/funTest/assets/projects/synthetic/npm/README.md [hunk 2]
```diff
@@ -1,26 +0,0 @@
-# npm
```
Comment (sschuberth) [inline plugins/package-managers/node/src/funTest/assets/projects/synthetic/npm/README.md:1]: Please still provide a rational in the commit message why it's ok to drop the contents of the file and why no useful information is lost.

**Comment by fviernau** [comment]
> Sorry, I need to get used to reading these in addition to the commit messages.

NP.

File: plugins/package-managers/node/src/funTest/assets/projects/synthetic/npm/README.md (hunk 2 above)
Comment (sschuberth) [reply plugins/package-managers/node/src/funTest/assets/projects/synthetic/npm/README.md:1]: Hint: Maybe it's easier to find an argument if you move the last commit to before this commit, and first add *all* links in this file to the respective places in the `NodePackageManager` enum. Then you could say that this file just describes the standard layout we have, which is somewhat self-explaining from the directory names, and that the links are by now maintained as part of class docs.

Comment (fviernau) [reply plugins/package-managers/node/src/funTest/assets/projects/synthetic/npm/README.md:1]: I tried to find a rationale for only keeping the shrinkwrap link

Comment (sschuberth) [reply plugins/package-managers/node/src/funTest/assets/projects/synthetic/npm/README.md:1]: > The other links are only for lockfile documentation.
As lockfile is common terminology and the links can be found easily,
it's not necessary to keep them around.

But the shrinkwrap file also is just a lockfile format... the [linked docs](https://docs.npmjs.com/cli/v6/configuring-npm/shrinkwrap-json) explicitly state:

> It is identical to `package-lock.json`, with one major caveat

So the current argumentation does not really hold: You should either keep all links to lockfile documentation, or none. I'd prefer to keep them all for easy lookup, and to avoid finding an argument to remove them.

Comment (fviernau) [reply plugins/package-managers/node/src/funTest/assets/projects/synthetic/npm/README.md:1]: > But the shrinkwrap file also is just a lockfile format... the [linked docs](https://docs.npmjs.com/cli/v6/configuring-npm/shrinkwrap-json) explicitly state:

I didn't know this was a lockfile and it wasn't obvious at all from the name to me. 

> So the current argumentation does not really hold:

I believe it does for above reason.

> I'd prefer to keep them all for easy lookup, and to avoid finding an argument to remove them.

If you still want these, what do you think would be a good place to put them ?

Comment (sschuberth) [reply plugins/package-managers/node/src/funTest/assets/projects/synthetic/npm/README.md:1]: Well, as [mentioned](https://github.com/oss-review-toolkit/ort/pull/9158#discussion_r1763176079) I would simply put the links next to corresponding `NodePackageManager` entries in the same way you did for shrinkwrap. So like:

    NPM(
        lockfileName = "package-lock.json", // See https://docs.npmjs.com/cli/v7/configuring-npm/package-lock-json.
        markerFileName = "npm-shrinkwrap.json" // See https://docs.npmjs.com/cli/v6/configuring-npm/shrinkwrap-json.
    ) {

    // ...

    PNPM(
        lockfileName = "pnpm-lock.yaml", // See https://pnpm.io/git#lockfiles.
        workspaceFileName = "pnpm-workspace.yaml"
    ) {

    // ...

    YARN(
        lockfileName = "yarn.lock" // See https://classic.yarnpkg.com/en/docs/yarn-lock.
    ) {

    // ..

    YARN2(
        lockfileName = "yarn.lock", // See  // See https://classic.yarnpkg.com/en/docs/yarn-lock..
        markerFileName = ".yarnrc.yml"
    ) {

Note that I've also added a link for PNPM's lockfile although it was not present in the `README.md` before for consistency.

**Review by sschuberth — APPROVED** [review]

**Comment by sschuberth** [comment]
> > Are you also planning to move Yarn2 and PNPM test assets below a common base directory?
> 
> Yes, I've explained this in the PR description.

Note to myself, the follow-up is at https://github.com/oss-review-toolkit/ort/pull/9183.
